apis/Google.Cloud.PubSub.V1/Google.Cloud.PubSub.V1/SubscriberClient.cs

```diff
@@ -1342,21 +1377,265 @@ private void StartPush()
                     _pushInFlight += acks.Count;
                     _concurrentPushCount += 1;
                     Task ackTask = _client.AcknowledgeAsync(_subscriptionName, acks, _hardStopCts.Token);
-                    Add(ackTask, Next(false, () => HandleAckResponse(ackTask, acks, null, null)));
+                    Add(ackTask, Next(false, _exactlyOnceDeliveryEnabled ? () => HandleAckResponseForExactlyOnce(ackTask, acks, null, null) : () => HandleAckResponse(ackTask, acks, null, null)));
                 }
                 if (extends.Count > 0)
                 {
                     _pushInFlight += extends.Count;
                     _concurrentPushCount += 1;
                     Task extendTask = _client.ModifyAckDeadlineAsync(_subscriptionName, extends.Select(x => x.Id), _modifyDeadlineSeconds, _hardStopCts.Token);
-                    Add(extendTask, Next(false, () => HandleAckResponse(extendTask, null, null, extends)));
+                    Add(extendTask, Next(false, _exactlyOnceDeliveryEnabled ? () => HandleAckResponseForExactlyOnce(extendTask, null, null, extends) : () => HandleAckResponse(extendTask, null, null, extends)));
                 }
                 if (nacks.Count > 0)
                 {
                     _pushInFlight += nacks.Count;
                     _concurrentPushCount += 1;
                     Task nackTask = _client.ModifyAckDeadlineAsync(_subscriptionName, nacks, 0, _hardStopCts.Token);
-                    Add(nackTask, Next(false, () => HandleAckResponse(nackTask, null, nacks, null)));
+                    Add(nackTask, Next(false, _exactlyOnceDeliveryEnabled ? () => HandleAckResponseForExactlyOnce(nackTask, null, nacks, null) : () => HandleAckResponse(nackTask, null, nacks, null)));
+                }
+            }
+
+            private void Retry(IEnumerable<string> temporaryFailures, Action<IEnumerable<string>> actionToRetry, int durationInSeconds = 600)
```

Comment:
Please rename to clarify what this is going to retry - and add a method comment for even more detail.